src/main/java/com/allocadia/carbonite/CarbonitePersist.java

```diff
@@ -0,0 +1,54 @@
+package com.allocadia.carbonite;
+
+import org.springframework.jdbc.core.support.JdbcDaoSupport;
+
+import com.allocadia.carbonite.utils.QueryUtils;
+import com.google.common.base.Joiner;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import lombok.SneakyThrows;
+
+public class CarbonitePersist<T extends PersistedObject> extends JdbcDaoSupport {
+
+    public Collection<T> save(Carbonite carbonite, Collection<T> objects) {
+        ModifiedObjects modifiedObjects = ModifiedObjects.from(objects);
+        Class<?> clazz = objects.iterator().next().getClass();
+        PersistenceInfo<PersistedObject> info = (PersistenceInfo<PersistedObject>) carbonite.getPersistenceInfo(clazz);
+        
+        persistNew(modifiedObjects.getNewObjects(), info);
+        //TODO: persistDirty
+    }
+
+    @SneakyThrows
+    private void persistNew(List<PersistedObject> newObjects, PersistenceInfo<PersistedObject> info) {
+        Map<Field, String> field2Column = info.getField2Column();
+        String baseSql = Joiner.on("\n").join(
+            "INSERT INTO " + info.getTableName(),
+            "(" + String.join(", ", field2Column.values()) + ")",
+            "VALUES",
+            "(" + QueryUtils.createParamList(field2Column.values().size()) + ")");
+        
+        List<Object[]> argList = new ArrayList<>();
+        for (PersistedObject persistedObject : newObjects) {
+            Object[] objectArgs = new Object[field2Column.keySet().size()];
+            
+            Iterator<Field> fieldIterator = field2Column.keySet().iterator();
+            int position = 0;
+            while (fieldIterator.hasNext()) {
+                objectArgs[position++] = fieldIterator.next().get(persistedObject);
+            }
+            
+            argList.add(objectArgs);
+        }
+        
+        super.getJdbcTemplate().batchUpdate(baseSql, argList);
```

Comment:
Past here I know I need to fetch the objects from the db to populate the ids...but I'm honestly not too sure how to do that.  I feel like doing something like fetching the newest N rows would only work with the assumption of a repeatable read isolation level, but maybe we'd want to write this in a way so that we could change the isolation level?  The only other way I can think to do this is basically to generate the ids on the java side.  Any ideas how to make this work with autoincrement?